src/mongocxx/private/collection.hpp

```diff
@@ -39,6 +39,19 @@ class collection::impl {
           client_impl(client) {
     }
 
+    impl(const impl& i) : 
+      collection_t{libmongoc::collection_copy(i.collection_t)},
+      database_name{i.database_name},
+      client_impl{i.client_impl}
+    {}
+
+    impl& operator=(const impl& i) { 
+        collection_t = libmongoc::collection_copy(i.collection_t);
```

Comment:
as with the database review, this is not a problem in practice as we only copy construct impls, even when copy assigning a `database_t`. I think that is cleaner than manually destroying the `collection_t`